eng/Version.Details.xml

```diff
@@ -4,6 +4,7 @@
     <Dependency Name="Microsoft.DotNet.Arcade.Sdk" Version="6.0.0-beta.21101.7">
       <Uri>https://github.com/dotnet/arcade</Uri>
       <Sha>2b430e5bbfaec37a6cead2f0cf79157f01f1a623</Sha>
+      <SourceBuild RepoName="arcade" ManagedOnly="true"/>
```

Comment:
@dagood Is arcade available as a source-build dep? I just realized that https://github.com/dotnet/sourcelink/blob/master/eng/Version.Details.xml doesn't have this.

Reply:
This line is should stay--the reason sourcelink can't depend on arcade is that arcade depends on sourcelink. There isn't always perfect logic in how we break the circular dependencies, this is just based on how we did it in .NET 5 and earlier, graphed out in the SVG at https://github.com/dotnet/source-build/blob/master/Documentation/planning/arcade-powered-source-build/implementation-plan.md

(If arcade wasn't available, I think the CI would fail to restore the `Microsoft.SourceBuild.Intermediate.arcade` package.)

Reply:
> the reason sourcelink can't depend on arcade is that arcade depends on sourcelink

This is actually interesting, in theory source-built sourcelink _can_ depend on source-built arcade, now that the work on arcade is done. Maybe we should actually end up doing that to be even more sure that source-built versions of arcade will work in that repo. In any case, that's separate from this PR: filed https://github.com/dotnet/source-build/issues/2022 to follow up.